gapis/api/vulkan/frame_loop.go

```diff
@@ -355,6 +365,28 @@ func (f *frameLoop) Transform(ctx context.Context, cmdId api.CmdID, cmd api.Cmd,
 				// Iterate through the loop contents, emitting instructions one by one.
 				for cmdIndex, cmd := range f.capturedLoopCmds {
 
+					switch cmd.(type) {
+					case *VkUnmapMemory:
+						vkCmd := cmd.(*VkUnmapMemory)
+						memID := vkCmd.Memory()
+						mem := GetState(out.State()).DeviceMemories().Get(memID)
+						addr := mem.MappedLocation().Address()
+						if addr != 0 {
+							// Only remember the first mapped target.
+							if _, ok := f.mappedAddress[addr]; !ok {
```

Comment:
As discussed offline-- we need to take care of when something is mapped and unmapped inside the loop. This code only seems to work when the map is before the loop start and the unmap is inside the loop.

Reply:
After further consideration, I'll split this CL into two. one deal with buffer create/destroy and the other for device memory. As your comments reminds me that the memory mapped also need to be unmapped,and the unmapped need to mapped.